src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/NpmBomTool.groovy

```diff
@@ -95,9 +97,16 @@ class NpmBomTool extends BomTool {
         } else if (containsPackageJson && containsNodeModules) {
             npmExePath = findExecutablePath(ExecutableType.NPM, true, detectConfiguration.getNpmPath())
             if (!npmExePath) {
-                logger.warn("Could not find a ${executableManager.getExecutableName(ExecutableType.NPM)} executable")
+                logger.warn("Could not find an ${executableManager.getExecutableName(ExecutableType.NPM)} executable")
             } else {
-                logger.debug("Npm version ${executableRunner.runExe(npmExePath, '-version').standardOutput}")
+                npmLsExe = new Executable(new File(sourcePath), npmExePath, ['-version'])
+                String npmNodePath = detectConfiguration.getNpmNodePath()
+                int lastSlashIndex = npmNodePath.lastIndexOf('/')
+                npmNodePath = npmNodePath.substring(0, lastSlashIndex)
+                if (npmNodePath) {
+                    npmLsExe.environmentVariables.put('PATH', npmNodePath)
```

Comment:
Is Node the only executable that NPM requires to be available on the PATH?

Reply:
In eclipse I replaced the path with an almost empty one and using these properties, npm ran just fine.